internal/command/import.go

```diff
@@ -236,11 +235,8 @@ func (c *ImportCommand) Run(args []string) int {
 	newState, importDiags := lr.Core.Import(lr.Config, lr.InputState, &terraform.ImportOpts{
 		Targets: []*terraform.ImportTarget{
 			{
-				Addr: addr,
-
-				// In the import block, the ID can be an arbitrary hcl.Expression,
-				// but here it's always interpreted as a literal string.
-				ID: hcl.StaticExpr(cty.StringVal(args[1]), configs.SynthBody("import", nil).MissingItemRange()),
+				LegacyAddr: addr,
+				IDString:   args[1],
```

Comment:
I considered adding "legacy" fields to `ImportTarget` in https://github.com/hashicorp/terraform/pull/33618 but decided on the HCL synth kludge instead. Glad we are removing it.